internal/cache/cache.go

```diff
@@ -6,14 +6,24 @@ import (
 )
 
 var (
-	ErrCacheNotFound    = errors.New("key not found in cache")
-	ErrCacheFailedToSet = errors.New("failed to set value in cache")
-	ErrCacheFailedToDel = errors.New("failed to delete value from cache")
-	ErrCacheFailedToGet = errors.New("failed to get value from cache")
+	ErrCacheNotFound          = errors.New("key not found in cache")
+	ErrCacheFailedToSet       = errors.New("failed to set value in cache")
+	ErrCacheFailedToDel       = errors.New("failed to delete value from cache")
+	ErrCacheFailedToGet       = errors.New("failed to get value from cache")
+	ErrCacheFailedToScan      = errors.New("failed to scan cache")
+	ErrCacheFailedToGetCount  = errors.New("failed to get count from cache")
+	ErrCacheFailedToExecuteTx = errors.New("failed to execute transaction")
 )
 
 type Store interface {
 	Get(key string) ([]byte, error)
 	Set(key string, value []byte, ttl time.Duration) error
-	Del(key string) error
+	Del(keys ...string) error
+
+	MapGet(hash string, key string) ([]byte, error)
```

Comment:
I would change args names:
`hash -> key|mapKey|mapID|hashsetKey|hashsetName`, `key->field|item`

Just suggestion, not a blocker

Reply:
Changed